climpred/graphics.py

```diff
@@ -230,3 +236,194 @@ def plot_bootstrapped_skill_over_leadyear(bootstrapped, plot_persistence=True, a
     ax.legend(frameon=False)
     ax.set_xlabel('Lead time [years]')
     return ax
+
+
+def check_only_climpred_dims(pe):
+    additional_dims = set(pe._datasets['initialized'].dims) - set(CLIMPRED_DIMS)
+    if len(additional_dims) != 0:
+        raise DimensionError(
+            f'{type(pe)}.plot() does not allow dimensions other '
+            f'than {CLIMPRED_DIMS}, found {additional_dims}. '
+            f'Please use .mean({additional_dims}) '
+            f'or .isel() before plot.'
+        )
+
+
+def plot_lead_timeseries_hindcast(
+    he, variable=None, ax=None, show_members=False, cmap='jet'
+):
+    """Plot datasets from HindcastEnsemble.
+
+    Args:
+        he (PredictionEnsemble): PredictionEnsemble.
+        variable (str or None): `variable` to plot. Defaults to the first in data_vars.
+        ax (plt.axes): axis to plot to. Defaults to False. Creates new.
+        show_members (bool): whether to display all members individually.
+            Defaults to True.
+        cmap (str): Name of plt colorbar. Defaults to 'tab10'.
+
+    Returns:
+        ax: plt.axes
+
+    """
+    check_only_climpred_dims(he)
+    if not variable:
+        variable = list(he._datasets['initialized'].data_vars)[0]
+    hind = he._datasets['initialized'][variable]
+    lead_freq = get_lead_cftime_shift_args(hind.lead.attrs['units'], 1)[1]
+    hist = he._datasets['uninitialized']
+    if isinstance(hist, xr.Dataset):
+        hist = hist[variable]
+    obs = he._datasets['observations']
+
+    cmap = mpl.cm.get_cmap(cmap, hind.lead.size)
+    if not ax:
+        _, ax = plt.subplots(figsize=(10, 4))
+    if isinstance(hist, xr.DataArray):
+        if 'member' in hist.dims and not show_members:
+            hist = hist.mean('member')
+            member_alpha = 1
+            lw = 2
+        else:
+            member_alpha = 0.5
+            lw = 1
+        hist.plot(
+            ax=ax,
+            lw=lw,
+            hue='member',
+            color='gray',
+            alpha=member_alpha,
+            label='uninitialized',
+        )
+
+    for i, lead in enumerate(hind.lead.values):
+        h = hind.sel(lead=lead).rename({'init': 'time'})
+        if not show_members and 'member' in h.dims:
+            h = h.mean('member')
+            lead_alpha = 1
+        else:
+            lead_alpha = 0.5
+        h['time'] = shift_cftime_index(h.time, 'time', int(lead), lead_freq)
+        h.plot(
+            ax=ax,
+            hue='member',
+            color=cmap(i),
+            label=f'initialized: lead={lead}',
+            alpha=lead_alpha,
+            zorder=hind.lead.size - i,
+        )
+
+    linestyles = ['-', ':', '-.', '--']
+    if len(obs) > 0:
+        for i, (obs_name, obs_item) in enumerate(obs.items()):
+            obs_item[variable].plot(
+                ax=ax,
+                color='k',
+                lw=3,
+                ls=linestyles[i],
```

Comment:
What if they have more than 4 observations? It seems crazy but it's possible. Right now it breaks with the error `IndexError: list index out of range`. So either start changing colors to shades of black or something. Or make an error that says this only supports 4 or less obs.

Reply:
Only found 4 ls options in plt

Reply:
Yeah I figured. I would just raise a custom error when you have obs > 4  and say we just support 4 obs plotting right now. I can't think right now of a better way to handle this that won't have many clashing colors.